feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/novacard/common/NovaCardRestrictionCheckMixin.kt

```diff
@@ -0,0 +1,73 @@
+package io.novafoundation.nova.feature_assets.presentation.novacard.common
+
+import io.novafoundation.nova.common.mixin.restrictions.RestrictionCheckMixin
+import io.novafoundation.nova.common.resources.ResourceManager
+import io.novafoundation.nova.common.view.bottomSheet.action.ActionBottomSheetLauncher
+import io.novafoundation.nova.common.view.bottomSheet.action.ButtonPreferences
+import io.novafoundation.nova.common.view.bottomSheet.action.primary
+import io.novafoundation.nova.feature_account_api.domain.interfaces.SelectedAccountUseCase
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.MultisigMetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.ProxiedMetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.hasChainAccountIn
+import io.novafoundation.nova.feature_account_api.domain.model.isThreshold1
+import io.novafoundation.nova.feature_assets.R
+import io.novafoundation.nova.runtime.ext.ChainGeneses
+
+private const val NOVA_CARD_AVAILABLE_CHAIN_ID = ChainGeneses.POLKADOT
+
+class NovaCardRestrictionCheckMixin(
+    private val accountUseCase: SelectedAccountUseCase,
+    private val resourceManager: ResourceManager,
+    private val actionLauncher: ActionBottomSheetLauncher,
+) : RestrictionCheckMixin {
+
+    override suspend fun isRestricted(): Boolean {
+        val selectedAccount = accountUseCase.getSelectedMetaAccount()
+        return isRestrictedByMultisig(selectedAccount) || isRestrictedByProxied(selectedAccount)
```

Comment:
I this the more structued approach would be to do exchaustive when:

``
fun isRestricted(metaAccount) = when(metaAccount) {
 is Multisig -> {..}
 is Proxied -> {..}
is ... -> false
}

This way we wont forget about this mixin when adding a new wallet type
And also we wont need to do additional `        if (metaAccount !is MultisigMetaAccount) return false
` checks inside each function